testing/tests/DevExpress.ui.widgets.htmlEditor/htmlEditorParts/api.tests.js

```diff
@@ -348,11 +348,23 @@ testModule('API', moduleConfig, () => {
         this.instance.option('value', 'New text');
         this.clock.tick();
 
-        assert.strictEqual(valueChangeStub.lastCall.args[0].value, 'New text');
+        assert.strictEqual(valueChangeStub.lastCall.args[0].value, '<p>New text</p>');
         assert.strictEqual(updateContentSpy.callCount, 2, 'value changed twice -> update content two times');
         assert.strictEqual(updateContentSpy.lastCall.args[0], 'New text', 'Update content with the new value');
     });
 
+    test('ValueChanged event should be triggered once when value contains non optimized markup(T1137577)', function(assert) {
+        this.createEditor();
+        const valueChangeStub = sinon.stub();
+
+        this.instance.on('valueChanged', valueChangeStub);
+        this.instance.option('value', '<p>after click</p><p></p>');
+        this.clock.tick();
+
+        assert.strictEqual(valueChangeStub.callCount, 1);
+        assert.strictEqual(valueChangeStub.getCall(0).args[0].value, '<p>after click</p>');
+    });
+
```

Comment:
Let's also add a test for case when we change value to another one, but new value's optimized version is the same as the previous one's. For example, `<p>before click</p>` -> `<p>before click</p><p></p>`.

Maybe there are some other possible cases?